evalml/pipelines/components/transformers/preprocessing/replace_nullable_types.py

```diff
@@ -0,0 +1,117 @@
+
+    Returns:
+        pandas.Series: The Dataframe column with the type changed to float64.
+
+    """
+    return series.astype("float64")
+
+
+def is_nullable_bool(df, col):
+    """Function to determine whether a column in a dataframe is a pandas NullableBoolean type.
+
+    Args:
+        df: Input dataframe to check.
+        col: Column of the input dataframe to check.
+
+    Returns:
+        boolean: Whether the column was of NullableBoolean type.
+    """
+    is_pandas_nullable = isinstance(df.dtypes[col], pca.boolean.BooleanDtype)
+    is_ww_nullable = isinstance(df.ww.logical_types[col], BooleanNullable)
+    return is_ww_nullable or is_pandas_nullable
+
+
+def replace_nullable_bool(series):
+    """Function to replace the pandas NullableBoolean column with a column of a compatible type.
+
+    Args:
+        series: The column to change the type of.
+
+    Returns:
+        pandas.Series: The Dataframe column with the type changed to category.
+
+    """
+    return series.astype("category")
+
+
+class ReplaceNullableTypes(Transformer):
+    """Transformer to replace features with the new nullable dtypes with a dtype that is compatible in EvalML."""
+
+    name = "Replace Nullable Types Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+
+    def __init__(self, random_seed=0, **kwargs):
+        parameters = {}
+        parameters.update(kwargs)
+
+        self._nullable_int_cols = []
+        self._nullable_bool_cols = []
+        super().__init__(
+            parameters=parameters, component_obj=None, random_seed=random_seed
+        )
+
+    def fit(self, X, y=None):
+        """Fits component to data.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features].
+            y (pd.Series, optional): The target training data of length [n_samples].
+
+        Returns:
+            self
+        """
+        X_t = infer_feature_types(X, ignore_nullable_types=True)
+        for col in X_t.columns:
+            if is_nullable_int(X_t, col):
+                self._nullable_int_cols.append(col)
+            elif is_nullable_bool(X_t, col):
+                self._nullable_bool_cols.append(col)
+                
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms data X by replacing columns that contain either the nullable integer or nullable boolean types
+        with the appropriate replacement type.  "float64" for nullable integers and "category" for nullable booleans.
+
+        Args:
+            X (pd.DataFrame): Data to transform
+            y (pd.Series, optional): Ignored.
+
+        Returns:
+            pd.DataFrame: Transformed X
+        """
+        X_t = infer_feature_types(X, ignore_nullable_types=True)
+        for col in self._nullable_int_cols:
+            X_t[col] = replace_nullable_int(X_t[col])
+        for col in self._nullable_bool_cols:
+            X_t[col] = replace_nullable_bool(X_t[col])
```

Comment:
I think we need to go through ww or else the schema will be invalidated. Here is an example from the unit tests where the logical and physical types don't match after transform and downstream components fail: 

![image](https://user-images.githubusercontent.com/41651716/143050793-136c8722-bb92-4af9-88c2-85257bc08457.png)

Reply:
I chatted with Tamar about the best way to do this and I think we should be good now.